src/models/beast.ts

```diff
@@ -102,6 +102,11 @@ export default class BeastModel {
         if (error instanceof Prisma.PrismaClientKnownRequestError) {
             if (isErrorWithTarget(error)) {
                 handleDatabaseError(error);
+                // Had to do this here, in the case where the query parameter (id for example)
+                // does not exist, it is a PrismaClientKnownRequestError, but does not have a target
+                // therefore the error does got handled smoothly, need to find a better way to handle this
+            } else if (error.code === 'P2025') {
```

Comment:
Maybe rather than do this we can refactor handleDatabaseError so that it can handle errors without the .target field. If the target field exists, add it to the error message, if not just return a generic error like "Not found"